homeassistant/components/stream/__init__.py

```diff
@@ -572,6 +573,19 @@ def get_diagnostics(self) -> dict[str, Any]:
         """Return diagnostics information for the stream."""
         return self._diagnostics.as_dict()
 
+    @staticmethod
+    async def async_has_stream_auth_error(hass: HomeAssistant, source: str) -> bool:
+        """Return true if rtsp stream raises an HTTPUnauthorizedError error."""
+        import av  # pylint: disable=import-outside-toplevel
+
+        pyav_options = {"rtsp_flags": "prefer_tcp", "timeout": "5000000"}
```

Comment:
What units are this? Seems pretty long.
Want to use `SOURCE_TIMEOUT`?

Also `CONF_PREFER_TCP`?

Reply:
I think it's seconds and I took it from this line here: https://github.com/home-assistant/core/blob/9c9e82a93e052431954e1908ca8ddc0268b470d8/homeassistant/components/stream/__init__.py#L158 but maybe it's that long because the `Stream` or `av` itself will kill the thread if it's own timeout is breached. I've updated it to `CONF_TIMEOUT` which is 30 seconds although `5` is still passed in to `av.open`.

`CONF_PREFER_TCP` done